app/services/teachers/manage.rb

```diff
@@ -13,32 +13,72 @@ def initialize(author:, teacher:, appropriate_body:)
     @appropriate_body = appropriate_body
   end
 
+  def self.find_or_initialize_by(trn:, trs_first_name:, trs_last_name:, event_metadata:)
+    teacher = Teacher.find_or_initialize_by(trn:) do |t|
+      t.trs_first_name = trs_first_name
+      t.trs_last_name = trs_last_name
+    end
+
+    if teacher.new_record? && teacher.save
```

Comment:
Commented on other PR asking about terseness vs verbosity on this occasion. Is it acceptable to splat event stuff because the lines above describe what it contains? Straw poll.

Reply:
Sounds good - I was going to tackle that in a subsequent PR. I'll raise a issue though and anyone's welcome to do it 🙂

Reply:
I was also going to create an `Events::Metadata` class we can use as it provides a touch more rigidity than a hash.